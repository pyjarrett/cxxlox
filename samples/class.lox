class Empty {}
print Empty;

var e = Empty();
print e;

class Vector {
    init(x, y) {
        this.x = x;
        this.y = y;
    }

    addVector(p) {
        this.x = this.x + p.x;
        this.y = this.y + p.y;
    }
    distance_to(p) {}
}
var p = Vector(10, 20);
print p;
print p.x;
print p.y;

var q = Vector(0, 0);
q.x = 30;
q.y = 40;

p.addVector(q);
print p.x;
print p.y;
