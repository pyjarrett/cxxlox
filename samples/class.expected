Empty
Empty instance
Vector instance
10
20
40
60
